src/lib/userAuthCall/transferFrom/index.ts

```diff
@@ -0,0 +1,141 @@
+import axios from "axios";
+import { BigNumber, providers } from "ethers";
+import { getAddress } from "ethers/lib/utils";
+
+import {
+  GELATO_RELAY_URL,
+  DEFAULT_DEADLINE_GAP,
+  getRelayAddress,
+  DEFAULT_INTERNAL_ERROR_MESSAGE,
+} from "../../../constants";
+import {
+  getEIP712Domain,
+  calculateDeadline,
+  getUserNonce,
+  signTypedDataV4,
+} from "../../../utils";
+import { PaymentType, RelayRequestOptions, RelayResponse } from "../../types";
+import { UserAuthSignature } from "../types";
+
+import {
+  EIP712_USER_AUTH_CALL_WITH_TRANSFER_FROM_TYPE_DATA,
+  UserAuthCallWithTransferFromPayloadToSign,
+  UserAuthCallWithTransferFromRequest,
+  UserAuthCallWithTransferFromRequestOptionalParameters,
+  UserAuthCallWithTransferFromStruct,
+} from "./types";
+
+const getPayloadToSign = (
+  struct: UserAuthCallWithTransferFromStruct
+): UserAuthCallWithTransferFromPayloadToSign => {
+  const verifyingContract = getRelayAddress(struct.chainId as number);
```

Comment:
you should start implementing a getRelayAddressWithWithTransferFrom even if it just return dumb address for now